project-base/app/src/FrontendApi/Resolver/Category/CategoryQuery.php

```diff
@@ -25,37 +38,94 @@ class CategoryQuery extends BaseCategoryQuery
+        private readonly AppFriendlyUrlFacade $appFriendlyUrlFacade,
+        private readonly ReadyCategorySeoMixFacade $readyCategorySeoMixFacade,
     ) {
         parent::__construct($categoryFacade, $domain, $friendlyUrlFacade);
     }
 
     /**
+     * {@inheritdoc}
+     *
+     * @deprecated This method is deprecated, use categoryOrSeoMixByUuidOrUrlSlugQuery instead
+     */
+    public function categoryByUuidOrUrlSlugQuery(?string $uuid = null, ?string $urlSlug = null): BaseCategory
+    {
+        throw new DeprecatedMethodException();
+    }
+
+    /**
+     * @param \GraphQL\Type\Definition\ResolveInfo $info
      * @param string|null $uuid
      * @param string|null $urlSlug
-     * @return \App\Model\Category\Category
+     * @return \App\Model\Category\Category|\App\Model\CategorySeo\ReadyCategorySeoMix
      */
-    public function categoryByUuidOrUrlSlugQuery(?string $uuid = null, ?string $urlSlug = null): Category
-    {
+    public function categoryOrSeoMixByUuidOrUrlSlugQuery(
+        ResolveInfo $info,
+        ?string $uuid = null,
+        ?string $urlSlug = null,
+    ): Category|ReadyCategorySeoMix {
         if ($uuid !== null) {
-            /** @var \App\Model\Category\Category $category */
-            $category = $this->getByUuid($uuid);
-
-            return $category;
+            return $this->getByUuid($uuid);
         }
 
         if ($urlSlug !== null) {
             $urlSlug = ltrim($urlSlug, '/');
 
-            /** @var \App\Model\Category\Category $category */
-            $category = $this->getVisibleOnDomainAndSlug($urlSlug);
+            $friendlyUrl = $this->appFriendlyUrlFacade->findByDomainIdAndSlug(
+                $this->domain->getId(),
+                $urlSlug,
+            );
+
+            if ($friendlyUrl === null) {
+                $modifiedSlug = TransformString::addOrRemoveTrailingSlashFromString($urlSlug);
+                $friendlyUrl = $this->appFriendlyUrlFacade->findByDomainIdAndSlug(
+                    $this->domain->getId(),
+                    $modifiedSlug,
+                );
+
+                if ($friendlyUrl === null) {
+                    throw new CategoryNotFoundUserError('Category with URL slug `' . $urlSlug . '` does not exist.');
+                }
+            }
+
+            $entityClass = $this->appFriendlyUrlFacade->getEntityClassByRouteName($friendlyUrl->getRouteName());
 
-            return $category;
+            if ($entityClass === Category::class) {
+                try {
+                    $category = $this->categoryFacade->getVisibleOnDomainById($this->domain->getId(), $friendlyUrl->getEntityId());
+                } catch (CategoryNotFoundException) {
+                    throw new CategoryNotFoundUserError('Category with URL slug `' . $urlSlug . '` does not exist.');
+                }
+
+                $matchingReadyCategorySeoMix = $this->findMatchingReadyCategorySeoMix($info, $category);
```

Comment:
This cannot happen, right? Slugs are unique so if it belongs to the category, we should display category.

Reply:
The requested behavior is – when you match the filter criteria with the existing SEO mix, you should be redirected to the appropriate SEO mix.